### PR TITLE
fix #158 - pass UnknownObjectException in issue_state

### LINKED ISSUE #158: update_bid_issues fails if an Issue is 404
Steps to reproduce:
1. Bid on an Issue
2. Change the issue url to something that will 404
3. Run `./manage.py update_bid_issues`

Expected results:
`update_bid_issues` should skip any 404 issues and continue updating others

Actual results:
```
(codesy)Macintosh-2:codesy owner$ ./manage.py update_bid_issuesTraceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/owner/projects/python_envs/codesy/lib/python2.7/site-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "/Users/owner/projects/python_envs/codesy/lib/python2.7/site-packages/django/core/management/__init__.py", line 377, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/owner/projects/python_envs/codesy/lib/python2.7/site-packages/django/core/management/base.py", line 288, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/Users/owner/projects/python_envs/codesy/lib/python2.7/site-packages/django/core/management/base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "/Users/owner/projects/codesy/codesy/auctions/management/commands/update_bid_issues.py", line 9, in handle
    update_issue_states()
  File "/Users/owner/projects/codesy/codesy/auctions/utils.py", line 50, in update_issue_states
    state = issue_state(issue.url)
  File "/Users/owner/projects/codesy/codesy/auctions/utils.py", line 25, in issue_state
    repo = github_client().get_repo(repo_name)
  File "/Users/owner/projects/python_envs/codesy/lib/python2.7/site-packages/github/MainClass.py", line 200, in get_repo
    "%s%s" % (url_base, full_name_or_id)
  File "/Users/owner/projects/python_envs/codesy/lib/python2.7/site-packages/github/Requester.py", line 169, in requestJsonAndCheck
    return self.__check(*self.requestJson(verb, url, parameters, headers, input, cnx))
  File "/Users/owner/projects/python_envs/codesy/lib/python2.7/site-packages/github/Requester.py", line 177, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.UnknownObjectException: 404 {u'documentation_url': u'https://developer.github.com/v3', u'message': u'Not Found'}
```

### PR DESCRIPTION
**Pull request by groovecoder**
If an issue has disappeared since first added to codesy, dont halt
the entire update_bid_issues batch on its UnkownObjectException.

TODO: log the exception somewhere

### CODE REVIEW

File: auctions/tests/utils_tests.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+        fake_gh_client = (fudge.Fake(github_client)
+                          .returns_fake()
+                          .provides('get_repo')
+                          .returns_fake()
+                          .provides('get_issue')
+                          .returns_fake()
+                          .has_attr(state='open'))
+
+        self.assertEqual('open', issue_state(url, fake_gh_client))
```
Comment (groovecoder) [inline auctions/tests/utils_tests.py]: @jsatt - can you help me debug fudge again?
```
Traceback (most recent call last):
  File "/Users/lcrouch/code/codesy/codesy/auctions/tests/utils_tests.py", line 13, in test_return_state
    fake_gh_client = (fudge.Fake(github_client)
  File "/Users/lcrouch/python/codesy/lib/python2.7/site-packages/fudge/__init__.py", line 1133, in returns_fake
    exp = self._get_current_call()
  File "/Users/lcrouch/python/codesy/lib/python2.7/site-packages/fudge/__init__.py", line 765, in _get_current_call
    "Call to a method that expects a predefined call but no such call exists.  "
FakeDeclarationError: Call to a method that expects a predefined call but no such call exists.  Maybe you forgot expects('method') or provides('method') ?
```

Comment (jsatt) [inline auctions/tests/utils_tests.py]: Not sure why I didn't get the email about this 6 days ago. 

The biggest issues here are what is returned from the various method calls. The docs aren't exactly clear and need to be updated. 

* `provides` doesn't allow you to dictate how it acts when called, it simply says it's there and could be called if it needs to be.
* `expects` says that this method is going to be called and allows you say dictate what args it's expected to be called with (`with_args`) and/or what to return when that happens (`returns` or `returns_fake`).
* additionally, `expects_call` says that the `Fake` itself is a method that expects to be called, and can then be chained with the same `with_args`, `returns` and `returns_fake` calls.
* `returns_fake` creates a new `Fake` instance. This is particularly important to remember when assigning a chained series to a variable. In the case above, the final `returns_fake` call after `get_issue` is the one being assigned to `fake_gh_client` not the original, so the instance you are injecting into the function call is not the one you expect.
* I'm not entirely sure what `fudge.Fake(github_client)` was trying to accomplish. The only arg that `Fake` accepts (others are deprecated) is a name, which is simply used to indicate to the developer which instance was involved in a failed test. There's no way of basing the `Fake` instance off of an existing class like you can with `Mock`'s `spec` arg.

This should be what you want.

```python
fake_gh_client = fudge.Fake()
(fake_gh_client
 .expects('get_repo').with_args('codesy/codesy').returns_fake()
 .expects('get_issue').with_args(158).returns_fake()
 .has_attr(state='open'))
```

Comment (groovecoder) [inline auctions/tests/utils_tests.py]: Great, thanks! You were exactly right - using `expects` allowed me to control the behavior how I wanted.

**Comment by groovecoder** [comment]
Merging this even though Travis output says it's not running any tests. :confused:

**Pull request merged**